### PR TITLE
Removed `supply` attribute from QDM types that no longer have it

### PR DESCRIPTION
**Pull request by hossenlopp**
The following types had `supply` removed in 5.4:

- Medication, Active
- Medication, Administered
- Substance, Administered
- Substance, Recommended
- Immunization, Administered

The following types have retained `supply` attribute:

- Medication, Discharge
- Medication, Dispensed
- Immunization, Order
- Medication, Order
- Substance, Order

Pull requests into CQL QDM Patient API require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1602 https://jira.mitre.org/browse/BONNIE-1659
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Code coverage has not gone down and all code touched or added is covered.
    * In rare situations, this may not be possible or applicable to a PR. In those situations:
        1. Note why this could not be done or is not applicable here: Drop in coverage percentage due to code being removed.
        2. Add TODOs in the code noting that it requires a test N/A
        3. Add a JIRA task to add the test and link it here: N/A
- [x] Automated regression test(s) pass

If JIRA tests were used to supplement or replace automated tests: N/A
- [x] JIRA test links: N/A
- [x] Justification for using JIRA tests: N/A
- [x] JIRA tests have been added to sprint

**Reviewer 1:**

Name: @edeyoung 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests: N/A
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name: @losborne
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests: N/A
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

File: spec/javascripts/datatypes/medication_spec.js.coffee [hunk 1]
```diff
@@ -233,7 +233,6 @@ describe "Medication", ->
       "start_time":1337846400,
       "statusOfMedication":null,
       "status_code":{"SNOMED-CT":["55561003"],"HL7 ActStatus":["active"]},
-      "supply":{"scalar":"1000","units":"g"},
```
Comment (edeyoung) [inline spec/javascripts/datatypes/medication_spec.js.coffee]: should not necessarily remove "supply" from the object as this is still supported in HDS and available in the UI. at a minimum, should include it as `null`.

Comment (hossenlopp) [reply spec/javascripts/datatypes/medication_spec.js.coffee]: Re-added supply to fixtures in 9c2503d.
